nbuild/02.build_manifests.md

```diff
@@ -0,0 +1,154 @@
+
+For example, the function [stdlib.fetch.fetch()](https://docs.raven-os.org/nbuild/master/stdlib.fetch.html#stdlib.fetch.fetch) can be used to download files (like the source code) before building a software. It looks for an entry named `fetch` in `versions_data` that contains the elements needed to download a file (like its URL, its hash, etc.).
+
+Here's an example of a `versions_data` for a build manifest supporting two versions (1.0.0 and 1.1.0) of the software, and uses the function [stdlib.fetch.fetch()`](https://docs.raven-os.org/nbuild/master/stdlib.fetch.html#stdlib.fetch.fetch) to download the source code:
+
+```
+[
+    {
+        'semver': '1.0.0',
+        'fetch': [{
+                'url': 'http://example.com/dl/hello_world_1.0.0.tar.gz',
+                'sha256': '5891b5b522d5df086d0ff0b110fbd9d21bb4fc7163af34d08286a2e846f6be03',
+            }
+        ],
+    },
+    {
+        'semver': '1.1.0',
+        'fetch': [{
+                'url': 'http://example.com/dl/hello_world_1.1.0.tar.gz',
+                'sha256': 'e258d248fda94c63753607f7c4494ee0fcbe92f1a76bfdac795c9d84101eb317',
+            },
+        ]
+    }
+]
+```
+
+## 2.2 Uploading packages to an official or unofficial Raven-OS server
+### 2.2.1 To an official Raven-OS server
+Building a package with `nbuild` has no incidence on the official repositories.
+To upload a package on one of Raven-OS official server, you will have to build it using [`build-raven-os-org`](https://build.raven-os.org), which will push the built packages on the repository [`unstable`](https://unstable.raven-os.org).
+The Raven-OS team will then take care of moving the packages considered safe enough to the repository [`beta`](https://beta.raven-os.org), and then to the repository [`stable`](https://stable.raven-os.org).
+
+### 2.2.2 To an unofficial Raven-OS server
+If you prefer not hosting your packages on one of Raven-OS offcial server but still want Raven-OS users to be able to use them, you can host your packages on your own instance of [Nest Server](https://github.com/raven-os/nest-server).
+
+## 2.3 `npf-checker`
+`npf-checker` is a checker that ensures the validity and conformance of an npf file.
+It will basically unpack it in a cache directory, fix the issues it found or let you fix them yourself, and repack it.
+It will be repacked as a new npf file called `[old_filename].new` and located in the same folder as the one you passed to `npf-checker`.
+
+As for its flags:
+* `-c` or `--cache-dir` to change the cache directory used by `npf-checker`, which is `cache/` by default.
+* `--visual` to use visual tools to fix the issues `npf-cheker` found yourself.
+It looks for the `visual` environment variable on your system and uses the software associated with it.
```

Comment:
Environment variables are usually in uppercase right?

Reply:
yep, my bad